### DISCUSSION TITLE
URL validation discussion

### DISCUSSION THREAD

**Discussion by abhijeetnishal** [abhijeetnishal/URLShortener#89] [General]
Discussion on URL validation before shortening

**Comment by abhijeetnishal** [comment]
Hey @RoshanAnsy, we can discuss here. Share you approach

**Comment by RoshanAnsy** [comment]
First, I would like to explain why this feature is important.

Let's assume a user visits your site and enters a URL. However, the entered URL might not be correct due to copying errors or other issues. After all that effort, the user might get a shortened URL that doesn't work when they try to use it. This experience could lead the user to think poorly of your site and decrease the likelihood of them returning because they perceive your URL shortening service as unreliable.

Therefore, we need this feature.

Approach 
1.Any kind of url when user inter it goes request to backend.
2.As the backend get the url from the frontend it route the request to middleware for validate url.
3.Now we  Set a User-Agent header to mimic a real web browser
4. We make a request with the user-entered URL along with headers. If it gets a status code 200, it means the URL is valid, and we pass the request to the next function for shortening the URL. Otherwise, we reject the request and send a message to the frontend stating, "URL is not valid.
